src/main/java/io/cdap/plugin/spark/SQLUtils.java

```diff
@@ -0,0 +1,218 @@
+    leavesSql.append(NAME_RESULTS_TEMP_TABLE).append(".").append(config.getLevelField()).append(", ");
+    for (String fieldName : nonMappedFields) {
+      leavesSql.append(NAME_RESULTS_TEMP_TABLE).append(".").append(fieldName).append(", ");
+    }
+    leavesSql.append("case when ").append(NAME_LEAVES_TEMP_TABLE).append(".").append(config.childField);
+    leavesSql.append(" is null then ").append("'").append(config.getFalseValueField()).append("'").append(" else ");
+    leavesSql.append("'").append(config.getTrueValueField()).append("' end as ").append(config.getBottomField());
+    leavesSql.append(" from ").append(NAME_RESULTS_TEMP_TABLE).append(" left join ").append(NAME_LEAVES_TEMP_TABLE);
+    leavesSql.append(" on ").append(NAME_RESULTS_TEMP_TABLE).append(".").append(config.childField).append("=");
+    leavesSql.append(NAME_LEAVES_TEMP_TABLE).append(".").append(config.childField);
+    return leavesSql.toString();
+  }
+
+  /**
+   * Generates sql statement to include additional fields (Level, Top, Bottom) with default values
+   *
+   * @param config {@link HierarchyToRelationalConfig}
+   * @param level  value for the level field
+   * @return {@link String} returns sql string for including additional fields
+   */
+  private static String sqlStatementForAdditionalFields(HierarchyToRelationalConfig config, int level) {
+    return String.format("%s as %s,'%s' as %s,'%s' as %s ", level, config.levelField, config.falseValueField,
+                         config.topField, config.falseValueField, config.bottomField);
+  }
+
+  public static String startQueryForRecursive(HierarchyToRelationalConfig config,
+                                              String tableName, String parentId) {
+    String additionalFields = sqlStatementForAdditionalFields(config, 1);
+    return String.format("select *, %s from %s where %s=%s", additionalFields, tableName,
+                         config.parentField, parentId);
+  }
+
+  /**
+   * Generates sql query for retrieving child records with parent ids that match the child id's from a temp table.
+   *
+   * @param conf          {@link HierarchyToRelationalConfig}
+   * @param count         level of the child fields
+   * @param treeTableName {@link String} name of the table containing the initial dataset (initial tree)
+   * @param tableName     {@link String} name of the table containing current level of records in the tree
+   * @return {@link String} sql statement for selecting child records based on given parent records
+   */
+  public static String recursiveQuery(HierarchyToRelationalConfig conf, int count,
+                                      String treeTableName, String tableName) {
+    String additionalFields = sqlStatementForAdditionalFields(conf, count);
+    return String.format("select indirect.*, %s FROM %s direct, %s indirect WHERE " +
+                           "direct.%s = indirect.%s", additionalFields.replace("0", String.valueOf(count + 1))
```

Comment:
if count is 10, would we replace 10 -> 111

Reply:
removed - it was leftover code that I missed